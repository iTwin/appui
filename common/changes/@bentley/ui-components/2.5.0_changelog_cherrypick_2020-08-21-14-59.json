{
  "changes": [
    {
      "comment": "",
      "type": "none",
      "packageName": "@bentley/ui-components"
    }
  ],
  "packageName": "@bentley/ui-components",
  "email": "Colin.Kerr@bentley.com"
}