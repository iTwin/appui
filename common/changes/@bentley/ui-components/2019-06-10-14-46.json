{
  "changes": [
    {
      "comment": "Tree: Fix an issue whith multiselection where shift-selecting nodes would not select anything.",
      "packageName": "@bentley/ui-components",
      "type": "none"
    }
  ],
  "packageName": "@bentley/ui-components",
  "email": "Robert.Lukasonok@bentley.com"
}