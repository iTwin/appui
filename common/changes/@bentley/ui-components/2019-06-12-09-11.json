{
  "changes": [
    {
      "comment": "Tree: Add ability to check or uncheck multiple selected nodes' checkboxes all at once.",
      "packageName": "@bentley/ui-components",
      "type": "none"
    }
  ],
  "packageName": "@bentley/ui-components",
  "email": "Robert.Lukasonok@bentley.com"
}