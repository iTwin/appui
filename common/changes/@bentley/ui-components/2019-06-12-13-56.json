{
  "changes": [
    {
      "comment": "BeInspireTree: Fix an issue with calling `updateNodesCheckboxes()` while handling `ModelLoaded` event",
      "packageName": "@bentley/ui-components",
      "type": "none"
    }
  ],
  "packageName": "@bentley/ui-components",
  "email": "Robert.Lukasonok@bentley.com"
}