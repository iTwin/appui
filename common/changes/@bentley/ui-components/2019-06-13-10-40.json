{
  "changes": [
    {
      "comment": "Tree: Correct TreeProps.checkboxInfo documentation.",
      "packageName": "@bentley/ui-components",
      "type": "none"
    }
  ],
  "packageName": "@bentley/ui-components",
  "email": "Robert.Lukasonok@bentley.com"
}