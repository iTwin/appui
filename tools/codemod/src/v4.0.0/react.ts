/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
import type { API, FileInfo, Options } from "jscodeshift";
import { useExtensions } from "../utils/Extensions";

export default function transformer(
  file: FileInfo,
  api: API,
  options: Options
) {
  const j = api.jscodeshift;
  useExtensions(j);

  const root = j(file.source);

  root.rename(
    "@itwin/appui-react:CommonWidgetProps",
    "@itwin/appui-react:Widget"
  );
  root.rename("@itwin/appui-react:WidgetProps", "@itwin/appui-react:Widget");

  return root.toSource(options.printOptions);
}
