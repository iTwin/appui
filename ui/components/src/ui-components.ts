/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
export { UiComponents } from "./ui-components/UiComponents";

export * from "./ui-components/breadcrumb/Breadcrumb";
export * from "./ui-components/breadcrumb/BreadcrumbPath";
export * from "./ui-components/breadcrumb/BreadcrumbTreeUtils";
export { BreadcrumbDragDropProps, withBreadcrumbDragDrop } from "./ui-components/breadcrumb/hoc/withDragDrop";
export * from "./ui-components/breadcrumb/breadcrumbdetails/BreadcrumbDetails";
export { BreadcrumbDetailsDragDropProps, withBreadcrumbDetailsDragDrop } from "./ui-components/breadcrumb/breadcrumbdetails/hoc/withDragDrop";

export { PageOptions } from "./ui-components/common/PageOptions";
export { SelectionMode } from "./ui-components/common/selection/SelectionModes";
export { SelectionHandler } from "./ui-components/common/selection/SelectionHandler";
export * from "./ui-components/common/showhide/ShowHideDialog";
export * from "./ui-components/common/showhide/ShowHideItem";
export * from "./ui-components/common/showhide/ShowHideMenu";

export * from "./ui-components/converters/TypeConverter";
export * from "./ui-components/converters/TypeConverterManager";
export * from "./ui-components/converters/BooleanTypeConverter";
export * from "./ui-components/converters/DateTimeTypeConverter";
export * from "./ui-components/converters/EnumTypeConverter";
export * from "./ui-components/converters/HexadecimalTypeConverter";
export * from "./ui-components/converters/NavigationPropertyTypeConverter";
export * from "./ui-components/converters/NumericTypeConverter";
export * from "./ui-components/converters/PointTypeConverter";
export * from "./ui-components/converters/StringTypeConverter";

import * as Primitives from "./ui-components/converters/valuetypes/PrimitiveTypes";
import * as ConvertedPrimitives from "./ui-components/converters/valuetypes/ConvertedTypes";
export { Primitives, ConvertedPrimitives };

export * from "./ui-components/dragdrop/DragDropDef";
export * from "./ui-components/dragdrop/withDragSource";
export * from "./ui-components/dragdrop/withDropTarget";
export * from "./ui-components/dragdrop/BeDragDropContext";

export * from "./ui-components/editors/EditorContainer";
export * from "./ui-components/editors/PropertyEditorManager";
export * from "./ui-components/editors/TextEditor";

export * from "./ui-components/filtering/FilteringInput";
export * from "./ui-components/filtering/ResultSelector";

export * from "./ui-components/properties/Description";
export * from "./ui-components/properties/EditorParams";
export * from "./ui-components/properties/Record";
export * from "./ui-components/properties/Value";
export * from "./ui-components/properties/ValueRendererManager";

export * from "./ui-components/properties/renderers/NonPrimitivePropertyRenderer";
export * from "./ui-components/properties/renderers/PrimitivePropertyRenderer";
export * from "./ui-components/properties/renderers/PropertyRenderer";
export * from "./ui-components/properties/renderers/PropertyView";

export * from "./ui-components/properties/renderers/label/NonPrimitivePropertyLabelRenderer";
export * from "./ui-components/properties/renderers/label/PrimitivePropertyLabelRenderer";

export * from "./ui-components/properties/renderers/value/PrimitivePropertyValueRenderer";
export * from "./ui-components/properties/renderers/value/ArrayPropertyValueRenderer";
export * from "./ui-components/properties/renderers/value/StructPropertyValueRenderer";
export * from "./ui-components/properties/renderers/value/DoublePropertyValueRenderer";
export * from "./ui-components/properties/renderers/value/NavigationPropertyValueRenderer";
export * from "./ui-components/properties/renderers/value/table/ArrayValueRenderer";
export * from "./ui-components/properties/renderers/value/table/StructValueRenderer";
export * from "./ui-components/properties/renderers/value/table/NonPrimitiveValueRenderer";

export * from "./ui-components/propertygrid/PropertyDataProvider";
export * from "./ui-components/propertygrid/SimplePropertyDataProvider";
export * from "./ui-components/propertygrid/component/PropertyGrid";
export * from "./ui-components/propertygrid/component/PropertyCategoryBlock";

export * from "./ui-components/table/TableDataProvider";
export * from "./ui-components/table/SimpleTableDataProvider";
export { Grid, GridProps } from "./ui-components/table/component/Grid";
export { Table, TableProps, TableSelectionTarget } from "./ui-components/table/component/Table";
export { TableCellEditorState, TableCellUpdatedArgs } from "./ui-components/table/component/Table";
export { TableDragDropType, TableDropTargetProps, TableDragDropProps, withTableDragDrop } from "./ui-components/table/hocs/withDragDrop";

export * from "./ui-components/tree/TreeDataProvider";
export * from "./ui-components/tree/HighlightingEngine";
export { Tree, TreeProps } from "./ui-components/tree/component/Tree";
export { TreeCellEditorState, TreeCellUpdatedArgs } from "./ui-components/tree/component/Tree";
export { TreeDragDropType, TreeDragDropProps, withTreeDragDrop } from "./ui-components/tree/hocs/withDragDrop";
export { SimpleTreeDataProvider, SimpleTreeDataProviderHierarchy } from "./ui-components/tree/SimpleTreeDataProvider";

export * from "./ui-components/viewport/ViewportComponent";
export * from "./ui-components/viewport/ViewportComponentEvents";

/** @docs-package-description
 * The ui-components package contains React components that are data-oriented, such as PropertyGrid, Table, Tree and Breadcrumb.
 * For more information, see [learning about ui-components]($docs/learning/components/index.md).
 */
/**
 * @docs-group-description Common
 * Common classes used across various UI components.
 */
/**
 * @docs-group-description Breadcrumb
 * Classes for working with a Breadcrumb.
 */
/**
 * @docs-group-description DragDrop
 * Classes and Higher Order Components for working with the DragDrop API.
 */
/**
 * @docs-group-description Filtering
 * Classes for working with filtering.
 */
/**
 * @docs-group-description Properties
 * Classes for working with Properties.
 */
/**
 * @docs-group-description PropertyEditors
 * Classes for working with Property Editors.
 */
/**
 * @docs-group-description PropertyGrid
 * Classes for working with a PropertyGrid.
 */
/**
 * @docs-group-description Table
 * Classes for working with a Table.
 */
/**
 * @docs-group-description Tree
 * Classes for working with a Tree.
 */
/**
 * @docs-group-description TypeConverters
 * Classes for working with Type Converters.
 */
/**
 * @docs-group-description Viewport
 * Classes for working with a Viewport.
 */
