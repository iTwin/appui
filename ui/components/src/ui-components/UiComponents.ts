/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @module Common */

import { I18N, TranslationOptions } from "@bentley/imodeljs-i18n";
import { UiError, getClassName } from "@bentley/ui-abstract";

/**
 * Entry point for static initialization required by various
 * components used in the package.
 * @public
 */
export class UiComponents {

  private static _i18n?: I18N;

  /**
   * Called by IModelApp to initialize the UiComponents
   * @param i18n The internationalization service created by the IModelApp.
   */
  public static async initialize(i18n: I18N): Promise<void> {
    UiComponents._i18n = i18n;
    await UiComponents._i18n.registerNamespace(UiComponents.i18nNamespace).readFinished;
    return Promise.resolve();
  }

  /** Unregisters the UiComponents internationalization service namespace */
  public static terminate() {
    if (UiComponents._i18n)
      UiComponents._i18n.unregisterNamespace(UiComponents.i18nNamespace);
    UiComponents._i18n = undefined;
  }

  /** The internationalization service created by the IModelApp. */
  public static get i18n(): I18N {
    if (!UiComponents._i18n)
      throw new UiError(UiComponents.loggerCategory(this), "UiComponents not initialized");
    return UiComponents._i18n;
  }

  /** The internationalization service namespace. */
  public static get i18nNamespace(): string {
    return "UiComponents";
  }

  /** @internal */
  public static get packageName(): string {
    return "ui-components";
  }

  /** Calls i18n.translateWithNamespace with the "UiComponents" namespace. Do NOT include the namespace in the key.
   * @internal
   */
  public static translate(key: string | string[], options?: TranslationOptions): string {
    return UiComponents.i18n.translateWithNamespace(UiComponents.i18nNamespace, key, options);
  }

  /** @internal */
  public static loggerCategory(obj: any): string {
    const className = getClassName(obj);
    const category = UiComponents.packageName + (className ? `.${className}` : "");
    return category;
  }

}
