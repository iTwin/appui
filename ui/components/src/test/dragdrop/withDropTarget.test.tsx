/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
import * as React from "react";
import { expect } from "chai";
import ReactTestUtils from "react-dom/test-utils";
import * as sinon from "sinon";
import { withDragSource, withDropTarget, DragSourceArguments } from "../../ui-components";
import TestBackend from "react-dnd-test-backend";
import { DragDropContext } from "react-dnd";
import { render, cleanup } from "react-testing-library";

describe("withDragSource", () => {

  /**
   * Wraps a component into a DragDropContext that uses the TestBackend.
   */
  function wrapInTestContext(DecoratedComponent: React.ComponentType<any>) {// tslint:disable-line:variable-name
    class TestContextContainer extends React.Component {
      public render() {
        return <DecoratedComponent {...this.props} />;
      }
    }

    return DragDropContext(TestBackend)(TestContextContainer);
  }

  class TestComponent extends React.Component<any> {
    public render(): React.ReactNode {
      return <div> test </div>;
    }
  }
  afterEach(cleanup);

  describe("Wrapped component", () => {
    const testDragSource = withDragSource(TestComponent);
    const BaseComponent = testDragSource.DecoratedComponent; // tslint:disable-line:variable-name
    it("mounts wrapped component", () => {
      render(<BaseComponent dragProps={{}} connectDragSource={(e: any) => e} />);
    });
  });
  describe("Drop functionality", () => {
    const testDropTarget = withDropTarget(TestComponent);
    const TestDragSource = withDragSource(testDropTarget); // tslint:disable-line:variable-name
    const ContextTestDragSource = wrapInTestContext(TestDragSource) as any; // tslint:disable-line:variable-name
    const onDragSourceBegin = sinon.spy((args: DragSourceArguments) => {
      args.dataObject = { test: true };
      return args;
    });
    const onDragSourceEnd = sinon.spy();
    const onDropTargetDrop = sinon.spy();
    const onDropTargetOver = sinon.spy();
    const canDropTargetDrop = sinon.spy(() => {
      return true;
    });
    const dragProps = { onDragSourceBegin, onDragSourceEnd, objectType: "test" };
    const dropProps = { onDropTargetDrop, onDropTargetOver, canDropTargetDrop, objectTypes: ["test"] };
    const root = ReactTestUtils.renderIntoDocument(<ContextTestDragSource dragProps={dragProps} dropProps={dropProps} />);

    // Obtain a reference to the backend
    const backend = (root as any).getManager().getBackend();
    const dragSource = ReactTestUtils.findRenderedComponentWithType(root as any, TestDragSource) as any;
    const dropTarget = ReactTestUtils.findRenderedComponentWithType(root as any, testDropTarget) as any;
    backend.simulateBeginDrag([dragSource.getHandlerId()]);
    // simulateHover must be called twice
    backend.simulateHover([dropTarget.getHandlerId()]);
    backend.simulateHover([dropTarget.getHandlerId()]);
    backend.simulateDrop();
    it("calls onDropTargetOver correctly", () => {
      expect(onDropTargetOver).to.have.been.calledOnce;
      expect(onDropTargetOver).to.have.been.calledWith(sinon.match({ dataObject: { test: true } }));
    });
    it("calls onDropTargetDrop correctly", () => {
      expect(onDropTargetDrop).to.have.been.calledOnce;
      expect(onDropTargetDrop).to.have.been.calledWith(sinon.match({ dataObject: { test: true } }));
    });
  });
});
