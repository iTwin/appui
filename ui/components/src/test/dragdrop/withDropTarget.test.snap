// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withDropTarget Wrapped component renders wrapped component correctly 1`] = `
<div>
  <TestComponent />
</div>
`;
